src/web-utils/videoplayer.ts

```diff
@@ -287,49 +287,23 @@ export class VideoPlayer {
 
   private _getVideoType(): boolean {
     let ret = false;
-    let vType = '';
     const sUrl: string = this._url ? this._url : '';
     if (sUrl != null && sUrl.length > 0) {
       try {
-        const val = sUrl.substring(sUrl.lastIndexOf('/')).match(/(.*)\.(.*)/);
-        if (val == null) {
-          vType = '';
+        const isHls = sUrl.match(/\.(m3u8)/i);
+        if (isHls) {
+          this._videoType = 'application/x-mpegURL';
+          ret = true;
         } else {
-          const a = sUrl.match(/(.*)\.(.*)/);
-          vType = a != null ? a[2].split('?')[0] : '';
-        }
-        switch (vType) {
-          case 'mp4':
-          case '':
```

Comment:
extention.blank appears to be missing from the new grep match.
This would allow an effective default to be mp4 and the "case default" to null only if the extension is specified and not found.

empty = mp4
not-empty-not-known = null

Any suggestions to do this would be fine as long a current/expected behavior is maintained.

Note: I apologize for not catching this the first round of review.